master/buildbot/status/web/base.py

```diff
@@ -403,6 +403,68 @@ def path_to_json_past_builds(request, builderName, number, filter_data=False):
     return url
 
 
+def get_query_branches_for_codebases(tags, codebases):
+    regex_branches = [
+        r'^(trunk)/',                 # Trunk
+        r'^(20[0-9][0-9].[0-9])\/',   # 2017.1/
+        r'^([0-9].[0-9])\/',          # 5.0/
+        r'^release\/([0-9].[0-9])/'   # release/4.6
+    ]
+
+    branches = map(lambda s: s.lower(), codebases.values())
+    codebase_keys = map(lambda s: s.lower(), codebases.keys())
+    query_branches = []
+
+    for branch, pattern in product(branches, regex_branches):
+        match = re.match(pattern, branch)
+        if match:
+            query_branches.append(match.group(1))
+
+    if not query_branches:
+        if 'unity' in codebase_keys:
+            query_branches = ['trunk']
+        else:
+            return []
+
+    if not filter(lambda tag: tag.split("-")[0].lower() in query_branches, tags):
+        query_branches = ['trunk']
+
+    return query_branches
+
+
+def filter_tags_by_codebases(tags, codebases):
+    """
+    Return list of tags filtered by branches from query params.
+
+    @param tags: list of builder-tags
+    @param codebases: dict with codebases from query, ex. {'unity': 'trunk', ...}
+    """
+    if not codebases:
+        return sorted(tags)
+
+    tag_as_branch_regex = r'^(20[0-9][0-9].[0-9]|[0-9].[0-9]|trunk)$'
```

Comment:
Same as other comment, are we duplicating this from somewhere else?

Reply:
Same as above.